TradeItIosTicketSDK2/TradeItPortfolioPositionPresenter.swift

```diff
@@ -1,63 +1,71 @@
+        guard let bidPrice = getQuote()?.bidPrice
+            else { return TradeItPresenter.MISSING_DATA_PLACEHOLDER }
+        return formatCurrency(bidPrice)
     }
     
-    func getFormattedSymbol() -> String {
-        return "N/A"
+    func formatCurrency(currency: NSNumber) -> String {
+                return NumberFormatter.formatCurrency(currency as Float)
     }
+    
+    func getFormattedAsk() -> String {
+        guard let askPrice = getQuote()?.askPrice
+            else { return TradeItPresenter.MISSING_DATA_PLACEHOLDER }
 
-    func getQuote() -> TradeItQuote? {
-        return self.tradeItPortfolioPosition.quote
+        return formatCurrency(askPrice)
     }
-
+    
     func getFormattedSpread() -> String {
-        guard let quote = getQuote() else { return "N/A" }
-        let high = quote.high as Float
-        let low = quote.low as Float
-        return formatCurrency(high - low)
-    }
+        guard let quote = getQuote()
+            , let high = quote.high as? Float
+            , let low = quote.low as? Float
+            else { return TradeItPresenter.MISSING_DATA_PLACEHOLDER }
 
-    func formatCurrency(currency: NSNumber) -> String {
-        return NumberFormatter.formatCurrency(currency as Float)
-    }
-
-    func getFormattedAsk() -> String {
-        guard let quote = getQuote() else { return "N/A" }
-        return formatCurrency(quote.askPrice)
+        return formatCurrency(high - low)
     }
+    
+    func getFormattedTotalValue() -> String {
+        guard let lastPrice = getQuote()?.lastPrice as? Float
+            , let quantity = getQuantity()
+            else { return TradeItPresenter.MISSING_DATA_PLACEHOLDER }
 
-    func getFormattedBid() -> String {
-        guard let quote = getQuote() else { return "N/A" }
-        return formatCurrency(quote.bidPrice)
-    }
+        let total = quantity * lastPrice
 
-    func getFormattedTotalValue() -> String {
-        guard let quote = getQuote() else { return "N/A" }
-        let total = getQuantity() * (quote.lastPrice as Float)
         return formatCurrency(total)
     }
 
     func getFormattedDayHighLow() -> String {
-        guard let quote = getQuote() else { return "N/A" }
-        let high = quote.high
-        let low = quote.low
-        return formatCurrency(low) + " - " + formatCurrency(high)
-    }
+        guard let quote = getQuote()
+            , let high = quote.high
+            , let low = quote.low
+        else { return TradeItPresenter.MISSING_DATA_PLACEHOLDER }
 
-    func getFormattedQuantity() -> String {
-        return "N/A"
+        return formatCurrency(low) + " - " + formatCurrency(high)
     }
+}
 
-    func getQuantity() -> Float {
-        return 0.0
+class TradeItPortfolioPositionPresenterFactory {
+    
+    static func forTradeItPortfolioPosition(tradeItPortfolioPosition: TradeItPortfolioPosition) -> TradeItPortfolioPositionPresenter {
+        if tradeItPortfolioPosition.position != nil {
```

Comment:
That's true -- is it possible for a `PortfolioPosition` to have both a `position` and an `fxPosition`?